src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -10,51 +10,157 @@
+        if event.entity_type == EventEntityType.SCENARIO or event.entity_type == EventEntityType.CYCLE:
+            self.scenarios = None
+            self.gui.broadcast(GuiCoreContext._CORE_CHANGED_NAME, {"scenario": True})
 
     def get_scenarios(self):
-        return []
+        if self.scenarios is None:
+
+            def add_scenarios(res, scens):
+                for scenario in scens:
+                    res.append((scenario.id, scenario.name, 1, scenario.is_primary, None))
+                return res
+
+            self.scenarios = []
+            for cycle, scenarios in tp.get_cycles_scenarios().items():
+                if cycle is None:
+                    add_scenarios(self.scenarios, scenarios)
+                else:
+                    self.scenarios.append((cycle.id, cycle.name, 0, False, add_scenarios([], scenarios)))
+        return self.scenarios
+
+    def get_scenario_configs(self):
+        if self.scenario_configs is None:
+            configs = tp.Config.scenarios
+            if isinstance(configs, dict):
+                self.scenario_configs = [(id, f"{c.id}") for id, c in configs.items()]
+        return self.scenario_configs
 
     def create_new_scenario(self, state: State, id: str, action: str, payload: t.Dict[str, str]):
-        pass
+        print(f"create_new_scenario(state, {id}, {action}, {payload}")
+        args = payload.get("args")
+        if args is None or not isinstance(args, list) or len(args) == 0 or not isinstance(args[0], dict):
+            return
+        data = args[0]
+        config_id = data.get(GuiCoreContext.__PROP_SCENARIO_CONFIG_ID)
+        scenario_config = tp.Config.scenarios.get(config_id)
+        if scenario_config is None:
+            state.assign(GuiCoreContext._ERROR_VAR, f"Invalid configuration id ({config_id})")
+            return
+        date_str = data.get(GuiCoreContext.__PROP_SCENARIO_DATE)
+        try:
+            date = parser.parse(date_str) if isinstance(date_str, str) else None
+        except Exception as e:
+            state.assign(GuiCoreContext._ERROR_VAR, f"Invalid date ({date_str}).{e}")
+            return
+        scenario = tp.create_scenario(scenario_config, date, data.get(GuiCoreContext.__PROP_SCENARIO_NAME))
+        if props := data.get("properties"):
+            with scenario as sc:
+                for prop in props:
+                    key = prop.get("key")
+                    if key and key not in GuiCoreContext.__SCENARIO_PROPS:
+                        sc._properties[key] = prop.get("value")
+        state.assign(GuiCoreContext._ERROR_VAR, "")
+
+    def select_scenario(self, state: State, id: str, action: str, payload: t.Dict[str, str]):
+        print(f"select_scenario(state, {id}, {action}, {payload}")
+        args = payload.get("args")
+        if args is None or not isinstance(args, list) or len(args) == 0 or not isinstance(args[0], dict):
+            return
+        data = args[0]
+        action = data.get("user_action")
```

Comment:
So that will ultimately be a 'change', right?

Reply:
yes, next iteration